lib/library/AssignLibraryPlugin.js

```diff
@@ -104,7 +104,8 @@ class AssignLibraryPlugin extends AbstractLibraryPlugin {
 	 * @returns {T | false} preprocess as needed by overriding
 	 */
 	parseOptions(library) {
-		const { name } = library;
+		const name = library.type === "commonjs2" ? null : library.name;
```

Comment:
another way is to add a parameter: `ignoreLibraryName`, `false` by default, for `commonjs2` it will be true:

```
case "commonjs2": {
  const AssignLibraryPlugin = require("./AssignLibraryPlugin");
  new AssignLibraryPlugin({
    type,
    prefix: ["module", "exports"],
    declare: false,
    ignoreLibraryName: true,
    unnamed: "assign"
  }).apply(compiler);
  break;
}
```